### PR TITLE
go 1.16.7

### PR DESCRIPTION
**Pull request by branchv**
Created with `brew bump-formula-pr`.

~`resource` blocks may require updates.~

### CODE REVIEW

**Comment by cho-m** [comment]
Let me restart CI since this is going to need long timeout and no fail fast.

**Comment by branchvincent** [comment]
Some overlap with #81919, but

Checksum errors:
* `baidupcs-go`
* `convox`
* `helmfile`
* `logcli`: #82894
* `promtail`: #82900

Test failures:
* `fn`
* `kapacitor`
* `orgalorg`
* `tunnel`

**Review by cuiweiqiang — COMMENTED** [review]
SHA256 checksum is same with offical website

**Comment by branchvincent** [comment]
closing in favor of `1.17` in #83413

File: Formula/go.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Go < Formula
   desc "Open source programming language to build simple/reliable/efficient software"
   homepage "https://golang.org"
-  url "https://golang.org/dl/go1.16.6.src.tar.gz"
-  mirror "https://fossies.org/linux/misc/go1.16.6.src.tar.gz"
-  sha256 "a3a5d4bc401b51db065e4f93b523347a4d343ae0c0b08a65c3423b05a138037d"
+  url "https://golang.org/dl/go1.16.7.src.tar.gz"
+  mirror "https://fossies.org/linux/misc/go1.16.7.src.tar.gz"
+  sha256 "1a9f2894d3d878729f7045072f30becebe243524cf2fce4e0a7b248b1e0654ac"
   license "BSD-3-Clause"
```
Comment (BHPettersen) [inline Formula/go.rb:7]: Ok

File: Formula/go.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
   head "https://go.googlesource.com/go.git"
```
Comment (BHPettersen) [inline Formula/go.rb:8]: Ok

File: Formula/go.rb [hunk 3]
```diff
@@ -1,9 +1,9 @@
 class Go < Formula
   desc "Open source programming language to build simple/reliable/efficient software"
   homepage "https://golang.org"
-  url "https://golang.org/dl/go1.16.6.src.tar.gz"
-  mirror "https://fossies.org/linux/misc/go1.16.6.src.tar.gz"
-  sha256 "a3a5d4bc401b51db065e4f93b523347a4d343ae0c0b08a65c3423b05a138037d"
+  url "https://golang.org/dl/go1.16.7.src.tar.gz"
+  mirror "https://fossies.org/linux/misc/go1.16.7.src.tar.gz"
+  sha256 "1a9f2894d3d878729f7045072f30becebe243524cf2fce4e0a7b248b1e0654ac"
```
Comment (BHPettersen) [inline Formula/go.rb:6]: Ok

File: Formula/go.rb (hunk 2 above)
Comment (BHPettersen) [inline Formula/go.rb:8]: Ok

**Review by BHPettersen — COMMENTED** [review]
Ok

**Comment by chipdude** [comment]
@branchvincent This should be reopened ... 1.17 includes changes that projects may not adopt for some time.

**Comment by SMillerDev** [comment]
It would be immediately overwritten by the 1.17 merge if we reopen this. If there is demand and it meets the requirements defined in https://docs.brew.sh/Versions someone can make a `go@1.16` version after 1.17 is merged.

**Comment by jameshalsall** [comment]
@chipdude which changes are you referring to in 1.17? toolchain?

**Pull request closed without merging**